framework/db/mysql/Schema.php

```diff
@@ -261,18 +261,42 @@ protected function findConstraints($table)
 AND rc.table_name = :tableName AND kcu.table_name = :tableName
 SQL;
 
-        $rows = $this->db->createCommand($sql, [':tableName' => $table->name])->queryAll();
-        $constraints = [];
-        foreach ($rows as $row) {
-            $constraints[$row['constraint_name']]['referenced_table_name'] = $row['referenced_table_name'];
-            $constraints[$row['constraint_name']]['columns'][$row['column_name']] = $row['referenced_column_name'];
-        }
-        $table->foreignKeys = [];
-        foreach ($constraints as $constraint) {
-            $table->foreignKeys[] = array_merge(
-                [$constraint['referenced_table_name']],
-                $constraint['columns']
-            );
+        try {
+            $rows = $this->db->createCommand($sql, [':tableName' => $table->name])->queryAll();
+            $constraints = [];
+            foreach ($rows as $row) {
+                $constraints[$row['constraint_name']]['referenced_table_name'] = $row['referenced_table_name'];
+                $constraints[$row['constraint_name']]['columns'][$row['column_name']] = $row['referenced_column_name'];
+            }
+            $table->foreignKeys = [];
+            foreach ($constraints as $constraint) {
+                $table->foreignKeys[] = array_merge(
+                    [$constraint['referenced_table_name']],
+                    $constraint['columns']
+                );
+            }
+        } catch (\Exception $e) {
+            $previous = $e->getPrevious();
+            if ($previous instanceof \PDOException && strpos($previous->getMessage(), 'SQLSTATE[42S02') !== false) {
```

Comment:
Maybe the condition can be reversed? This saves one level of indentation:
~~~php
if (!$previous instanceof \PDOException || strpos($previous->getMessage(), 'SQLSTATE[42S02') === false) {
    throw $e;
}
~~~